aws-api/src/main/java/com/amplifyframework/api/aws/AWSApiPlugin.java

```diff
@@ -732,4 +748,43 @@ public int hashCode() {
             return result;
         }
     }
+
+    /**
```

Comment:
In a microservice I did a couple years back, I found a pattern where you can add _behaviors_ to an `enum`. In that case, I would do like `Metrics.SOME_SERVICE_CRAP_HAPPENED.report()`.

But, you could use a similar mechanism here, potentially. Maybe something like:
```java
ApiEndpointStatus.UNKNOWN.announce();
```
And then have the hub broadcast logic contained inside the enum itself.